File: practica2_844759_845647/java/Add.java
//------------------------------------------------------------------------------
// File:   add.java
// Authors: Jorge Leris - 845647 / Angel Villanueva - 844759
// Date:   febrero 2023
// Coms:   Práctica 2 de TecPro
//------------------------------------------------------------------------------

import java.util.Stack;

public class Add extends Instruccion
{
    public void ejecutar(Stack<Integer> Pila, int contador){
        int num1 = Pila.peek();
        pila.pop();
        int num2 = Pila.peek();
        Pila.pop();
        Pila.push(num1 + num2);
        contador = contador + 1;
    }

    public void listar(){
        System.out.println("add");
    }
}

File: practica2_844759_845647/c++/add.cc
//------------------------------------------------------------------------------
// File:   add.cc
// Authors: Jorge Leris - 845647 / Angel Villanueva - 844759
// Date:   febrero 2023
// Coms:   Fichero "add" de la práctica 2
//------------------------------------------------------------------------------
#include "add.h"

void Add::ejecutar(stack<int> &pila, int &contador) const
{
    int num1 = pila.top();
    pila.pop();
    int num2 = pila.top();
    pila.pop();
    pila.push(num1 + num2);
    contador++;
}

void Add::listar() const
{
    cout << "add" << endl;
}